lib/Interpreter/Interpreter.cpp

```diff
@@ -265,6 +275,30 @@ namespace cling {
             } else
               cling::errs() << Sym << " not in Module!\n";
           }
+
+          const clang::Decl* Scope =
+              LangOpts.CPlusPlus
+                  ? m_LookupHelper->findScope("cling::runtime",
+                                              LookupHelper::NoDiagnostics)
+                  : CI->getSema().getASTContext().getTranslationUnitDecl();
+          if (!Scope) {
+            cling::errs() << "Scope for gCling was not found\n";
+          }
+          else if (const clang::ValueDecl* gCling =
+              m_LookupHelper->findDataMember(Scope, "gCling",
+                                             LookupHelper::NoDiagnostics)) {
+            using namespace utils;
+            const std::string Name = Analyze::maybeMangleDeclName(gCling);
+            if (!Name.empty()) {
+              // gCling gets linked to top-most Interpreter.
```

Comment:
If I read the code correctly, the code below does not implement the comment (since m_Parenting[0] is always set to 'this' rather than the parent.  
In addition, I fail to see/understand the code that would prevent the symbol to be added multiple time (one per interpreter).

What am I missing?